inngest-spring-boot-demo/src/test/java/com/inngest/springbootdemo/RateLimitedFunctionIntegrationTest.java

```diff
@@ -0,0 +1,46 @@
+package com.inngest.springbootdemo;
+
+import com.inngest.Inngest;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.parallel.Execution;
+import org.junit.jupiter.api.parallel.ExecutionMode;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@IntegrationTest
+@Execution(ExecutionMode.CONCURRENT)
+class RateLimitedFunctionIntegrationTest {
+    @Autowired
+    private DevServerComponent devServer;
+
+    @Autowired
+    private Inngest client;
+
+    @Test
+    void testFunctionIsRateLimited() throws Exception {
+        String event1 = InngestFunctionTestHelpers.sendEvent(client, "test/rateLimit").getIds()[0];
+        // Rate limit test function is limited to 2 over 6 seconds. Based on the simplistic description of GCRA in
+        // https://www.inngest.com/docs/guides/rate-limiting#how-rate-limiting-works
+        // we need to sleep at least 3 seconds here for the second event not to get rate limited
```

Comment:
This test was buggy before instead of just being flaky. After reading the documentation at https://www.inngest.com/docs/guides/rate-limiting#how-rate-limiting-works, I adjusted the sleep times for the test to hopefully run more robustly.